Ingest modeled parameters sanghi23

kaseyLee123 opened the pull request:
Short description: Include what type of data being ingested and appropriate references.

Link to relevant issue: Closes #518 and Closes #484 .

For data ingests:
[ ] includes script used for ingest
[ ] includes modified JSON files
[ ] Add new tests
[ ] Update the Versions table

kelle commented on data/reference/Parameters.json:
```diff
@@ -26,5 +26,9 @@
+    },
+    {
+        "parameter": "L bol",
+        "description": "l bol"
```
```suggestion
        "description": "bolometric luminosity"
```

kelle commented:
Also update with main.  It shouldn't impact this PR since it's all photometry and proper motions.

kaseyLee123 commented:
I just edited pytest cases where I added one for checking which had a null value for model and also how many of each parameter was ingested (added one for L bol). I added counters in my script and have the number in the comments next to it and the pytest case numbers make sense when I do the math

kelle commented on tests/test_integrity.py:
```diff
@@ -423,6 +423,16 @@ def test_modeled_parameters(db):
         print(t)
     assert len(t) == 0
 
+    t = (
+        db.query(db.ModeledParameters)
+        .filter(db.ModeledParameters.c.model.is_(None))
+        .astropy()
+    )
+    if len(t) > 0:
+        print("\nEntries found without a model")
+        print(t)
+    assert len(t) == 705
+
```
This is not a data "integrity" tests like the other ones that are here. Make a new `test_data_modeled_params.py` file for this test.  It's totally fine for modeled parameter entries to not have a model.

kaseyLee123 commented:
Just pushed the new pytest file. I have the JSON files (a little over 1000) ready to push whenever

kaseyLee123 commented:
> Awesome progress!
> 
> * could you please try to push just 10-20 JSON files.  If you're still using Github Desktop, use the checkboxes to Deselect All and then Select 10-20 by hand to add, commit, and push.
> * add tests which test the references for the modeled paramaters. How many Lbols from Fili15?

Did you want the pytest case to check references for each parameter or just in general?

kelle commented:
For each parameter.  How many Lbols from Fili15? How many Lbols from Sang24?

kelle commented on scripts/ingests/sanghi23/sanghi23_modeled_parameters_skipped.csv:
```diff
@@ -0,0 +1,16 @@
+# delimiter: ','
+# schema: astropy-2.0
+Skipped,Reason
+veryred-1527,"no source found, atmo model"
```
Is there really a source name called "veryred-1527"?  @cosmicoder, any insight on what source this is?

kaseyLee123 commented:
just did! Not sure if the pytest case is the most efficient way because it seems a little long but. Also, the csv files did not have a ref column so I just put sang23 for all of them.

kelle commented:
Yep, this looks great.  Once you ingest the one skipped source (PSO J243.9421+67.2075), go ahead and push the JSONs!

kaseyLee123 commented:
just pushed

kelle reviewed (approved):
I think this is done. Great work!!!

kelle commented:
Ohhhh, adding the adopted column!  Let's go ahead and do that now since you're already modifying the schema.
- Add the column - set them all to Null or 0.
- We don't have to actually adopt any measurements at the moment -- that will be hard to decide!
- Add a test which counts the number of adopted measurements. Will be zero for this PR.

kelle commented:
Also, take a look at the [ModeledParameters ingest template spreadsheet](https://docs.google.com/spreadsheets/d/1NsfTozaSDUiaM-uE4kJVs42KXc_n3Uj1Q4UtqCgswVE/edit?gid=0#gid=0).  Let's make sure what you're doing here is reflected in that template.

kelle commented:
Looking at the ingest template, let's also add the `other_references` column.

kaseyLee123 commented:
Just did everything looks good I think

kelle reviewed (approved):
Amazing fast work!  You're knocking it out of the park!
